src/main/java/unicash/logic/parser/FindCommandParser.java

```diff
@@ -1,36 +1,127 @@
 package unicash.logic.parser;
 
+import static java.util.Objects.requireNonNull;
 import static unicash.logic.UniCashMessages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static unicash.logic.parser.CliSyntax.PREFIX_AMOUNT;
+import static unicash.logic.parser.CliSyntax.PREFIX_CATEGORY;
+import static unicash.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static unicash.logic.parser.CliSyntax.PREFIX_LOCATION;
+import static unicash.logic.parser.CliSyntax.PREFIX_NAME;
+import static unicash.logic.parser.CliSyntax.PREFIX_TYPE;
 
-import java.util.Arrays;
-
+import unicash.commons.util.ToStringBuilder;
 import unicash.logic.commands.FindCommand;
 import unicash.logic.parser.exceptions.ParseException;
-import unicash.model.transaction.predicates.TransactionContainsKeywordsPredicate;
+import unicash.model.category.Category;
+import unicash.model.commons.Amount;
+import unicash.model.transaction.DateTime;
+import unicash.model.transaction.Location;
+import unicash.model.transaction.Name;
+import unicash.model.transaction.Type;
+import unicash.model.transaction.predicates.TransactionContainsAllKeywordsPredicate;
+
 
 /**
  * Parses input arguments and creates a new FindCommand object
  */
 public class FindCommandParser implements Parser<FindCommand> {
 
+    private TransactionContainsAllKeywordsPredicate findPredicate;
+
+    /**
+     * Creates a {@code FindCommandParser} object with a default
+     * {@code TransactionContainsAllKeywordsPredicate} predicate.
+     */
+    FindCommandParser() {
+        findPredicate = new TransactionContainsAllKeywordsPredicate();
+    }
+
+
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
+        requireNonNull(args);
```

Comment:
Actually abt this, I think the assert is for the course scripts to check if we use assert, for some reason they are checking for the usage of assert in our code that's why it's like inserted everywhere recently by everyone, is it possible if we keep both, since requireNonNull works regardless of assertions being enabled in our build file so it's actually useful for our code and unit tests, and `assert` makes the scripts that check our code happy

Reply:
I don't think it's one of the milestones this time right? I don't think it's good to include both since one running means the other is never run. But if you mean keeping the 2 variations, sure but I'm not super keen on it, but if it's to appease the script, sure